### PR TITLE
API test use webdav version table

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds `Scenario Outline` table for using `old DAV path` and `new DAV path` in a single scenario.

## How Has This Been Tested?
Locally

## Motivation
To make reading test scenarios more easier.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by paurakhsharma** [comment]
Some problem with caldav-test. Can someone restart the drone?

File: tests/acceptance/features/apiMain/checksums.feature [hunk 1]
```diff
@@ -108,16 +108,6 @@ Feature: checksums
     Then the header checksum should match "SHA1:3ee962b839762adb0ad8ba6023a4690be478de6f"
 
   @local_storage
```
Comment (phil-davis) [inline tests/acceptance/features/apiMain/checksums.feature]: Looks like this tag should also be deleted.

File: tests/acceptance/features/apiMain/favorites.feature [hunk 2]
```diff
@@ -165,23 +137,20 @@ Feature: favorite
 		And user "user1" gets the following properties of file "/favoriteFile.txt" using the API
 			|{http://owncloud.org/ns}favorite|
 		Then the single response should contain a property "{http://owncloud.org/ns}favorite" with value "0"
+		Examples:
+			| dav_version   |
+			| old           |
+			| new           |
 
-	Scenario: sharee file favorite state should not change the favorite state of sharer
```
Comment (phil-davis) [inline tests/acceptance/features/apiMain/favorites.feature]: This scenario is different/opposite to the one above. I think it needs to come back and have scenario outline applied.

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+			| old           |
+			| new           |
+
+	Scenario Outline: A disabled user cannot use webdav
+		Given using <dav_version> DAV path
+		And user "userToBeDisabled" has been created
+		And user "userToBeDisabled" has been disabled
+		When user "userToBeDisabled" downloads the file "/welcome.txt" using the API
+		Then the HTTP status code should be "401"
+	Examples:
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: indent

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+		When user "user0" moves file "/textfile0.txt" to "/FOLDER/textfile0.txt" using the API
+		Then user "user0" file "/FOLDER/textfile0.txt" should have the previously stored id
+		And user "user0" should not see the following elements
+			| /textfile0.txt |
+		Examples:
+			| dav_version   |
+			| old           |
+			| new           |
+
+	Scenario Outline: Renaming a folder to a backslash encoded should return an error using new endpoint
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: delete "using new endpoint" from the scenario outline descriptions (from here down...)

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+		When user "user0" moves folder "/testshare" to "/not-existing/testshare" using the API
+		Then the HTTP status code should be "409"
+		And user "user0" should see the following elements
+			| /testshare/ |
+		Examples:
+			| dav_version   |
+			| old           |
+			| new           |
+
+	Scenario Outline: Downloading a file on the new endpoint should serve security headers
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: delete "on the new endpoint"

File: tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature [hunk 6]
```diff
@@ -3,463 +3,7 @@ Feature: webdav-related-new-endpoint
...
-		And user "user0" has created a share with settings
-			| path        | testcustompropshared.txt |
-			| shareType   | 0                        |
-			| permissions | 31                       |
-			| shareWith   | user1                    |
-		And user "user0" has set property "{http://whatever.org/ns}very-custom-prop" of file "/testcustompropshared.txt" to "valueForSharetest"
-		When user "user1" gets a custom property "{http://whatever.org/ns}very-custom-prop" of file "/testcustompropshared.txt"
-		Then the response should contain a custom "{http://whatever.org/ns}very-custom-prop" property with "valueForSharetest"
-
-	Scenario: Setting custom DAV property using a new endpoint and reading it using an old endpoint
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature]: This scenario has gone nowhere. It is a mix of old and new. Maybe we should make a feature called ``webdav-related-old-vs-new.feature`` ?
There is a "requirement"  that when something is done using new DAV path, that the change is seen when using old DAV path and the opposite.
You could even write the scenario with an outline table like:
```
	Examples:
			| action_dav_version | other_dav_version |
			| old | new|
			| new | old |
```

File: tests/acceptance/features/apiWebdav/webdav-related-old-endpoint.feature [hunk 7]
```diff
@@ -3,442 +3,6 @@ Feature: webdav-related-old-endpoint
...
-		And user "user0" has created a share with settings
-			| path        | testcustompropshared.txt |
-			| shareType   | 0                        |
-			| permissions | 31                       |
-			| shareWith   | user1                    |
-		And user "user0" has set property "{http://whatever.org/ns}very-custom-prop" of file "/testcustompropshared.txt" to "valueForSharetest"
-		When user "user1" gets a custom property "{http://whatever.org/ns}very-custom-prop" of file "/testcustompropshared.txt"
-		Then the response should contain a custom "{http://whatever.org/ns}very-custom-prop" property with "valueForSharetest"
-
 	Scenario: Setting custom DAV property using an old endpoint and reading it using a new endpoint
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related-old-endpoint.feature]: See comment above - this scenario can probably be made into a scenario outline...

**Review by phil-davis — CHANGES_REQUESTED** [review]
There are a few changes to make - see comments.
But I think this looks like a good thing to do. It removes all the copied scenarios, condensing them into scenario outlines.

**Comment by phil-davis** [comment]
@paurakhsharma 
1) I rebased this to fix the caldav issue (there are now 4 caldav/carddav test jobs on drone and a folder got renamed). So you will need to delete your local branch and get it again from GitHub.
2) Please keep the first commit, and add a commit with the changes - that will be much easier to review.
3) When it is all good, it can be squashed.

**Comment by phil-davis** [comment]
And codecov results did not come. Don't worry about that - might be luckier when you push another commit :)

File: tests/acceptance/features/apiWebdav/webdav-related-old-endpoint.feature (hunk 7 above)
Comment (paurakhsharma) [reply tests/acceptance/features/apiWebdav/webdav-related-old-endpoint.feature]: I have moved scenario to webdav-related.feature

File: tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature (hunk 6 above)
Comment (paurakhsharma) [reply tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature]: Thank you for this ides.
I have moved scenario to webdav-related.feature

File: tests/acceptance/features/apiMain/checksums.feature [hunk 8]
```diff
@@ -156,7 +155,6 @@ Feature: checksums
 			| old           |
 			| new           |
 
-  @local_storage
```
Comment (phil-davis) [inline tests/acceptance/features/apiMain/checksums.feature]: This looks like it uses local_storage - so the tag should stay.

File: tests/acceptance/features/apiMain/favorites.feature (hunk 2 above)
Comment (phil-davis) [reply tests/acceptance/features/apiMain/favorites.feature]: This still needs to come back.

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 9]
```diff
@@ -558,27 +558,18 @@ Feature: webdav-related
 			| old           |
 			| new           |
 
-	Scenario Outline: Doing a GET with a web login should work with CSRF token on the new backend
+	Scenario: Doing a GET with a web login should work with CSRF token on the new backend
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: Does this really only work on the new backend?
Or can you put back the scenario outline, and just change the scenario description?

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 10]
```diff
@@ -558,27 +558,18 @@ Feature: webdav-related
...
 		When the client sends a "GET" to "/remote.php/dav/files/user0/welcome.txt" with requesttoken using the API
 		Then the downloaded content should start with "Welcome to your ownCloud account!"
 		And the HTTP status code should be "200"
-		Examples:
-			| dav_version   |
-			| old           |
-			| new           |
 
-	Scenario Outline: Doing a PROPFIND with a web login should work with CSRF token on the new backend
+	Scenario: Doing a PROPFIND with a web login should work with CSRF token on the new backend
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: Does this really only work on the new backend?
Or can you put back the scenario outline, and just change the scenario description?

File: tests/acceptance/features/apiWebdav/webdav-related.feature [hunk 11]
```diff
@@ -622,3 +613,16 @@ Feature: webdav-related
 			| dav_version   |
 			| old           |
 			| new           |
+
+	Scenario Outline: Setting custom DAV property using one endpoint and reading it with other endpoint
```
Comment (phil-davis) [inline tests/acceptance/features/apiWebdav/webdav-related.feature]: good

File: tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature (hunk 6 above)
Comment (phil-davis) [reply tests/acceptance/features/apiWebdav/webdav-related-new-endpoint.feature]: good

**Review by phil-davis — COMMENTED** [review]
A couple more comments...

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@DeepDiver1975 or @PVince81 - lost the codecov battle here. This is Behat feature files only, so I have no idea what codecov is thinking.

Can you override and merge this please?

**Comment by paurakhsharma** [comment]
Backport on #31842

**Pull request merged**